File: 数仓表结构/asset_package/src/main/java/com/weshare/interfaces/AssetFiles.java
package com.weshare.interfaces;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.SparkSession;

/**
 * Created by mouzwang on 2020-11-30 15:41
 */
public interface AssetFiles {

    void insertData(Dataset<String> dataset,SparkSession sparkSession,String fileId);

    void updateData(Dataset<String> dataset,SparkSession sparkSession,String fileId);

    void deleteData(Dataset<String> dataset, SparkSession sparkSession,String fileId);
}

File: 数仓表结构/asset_package/src/main/scala/com/weshare/interface/AssetFiles.scala
package com.weshare.interface

import org.apache.spark.sql.{Dataset, SparkSession}

/**
  * @author ximing.wei 2021-05-27 14:45:20
  */
trait AssetFiles {

    def insertData(sparkSession: SparkSession, dataset: Dataset[String], projectId: String)

    def updateData(sparkSession: SparkSession, dataset: Dataset[String], projectId: String)

    def deleteData(sparkSession: SparkSession, dataset: Dataset[String], projectId: String, importId: String)
}
